app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -97,13 +105,24 @@ class ParkingViewModel(
         Point.fromLngLat(
             maxOf(startPos.longitude(), endPos.longitude()),
             maxOf(startPos.latitude(), endPos.latitude()))
-
-    Log.d("ParkingViewModel", "Getting parkings between $startPos and $endPos")
-    parkingRepository.getParkingsBetween(
-        bottomLeft,
-        topRight,
-        { _rectParkings.value = it },
-        { Log.e("ParkingViewModel", "Error getting parkings: $it") })
+    // Get all tiles that are in the rectangle
+    tilesToDisplay = Tile.getAllTilesInRectangle(bottomLeft, topRight)
+    tilesToDisplay.forEach { tile ->
+      if (tilesToParking.value.containsKey(tile)) {
+        _rectParkings.value += tilesToParking.value[tile]!!
+        return@forEach // Skip to the next tile if already fetched
+      }
+      tilesToParking.value[tile] = emptyList() // Avoid querying the same tile multiple times
+      parkingRepository.getParkingsBetween(
+          tile.bottomLeft,
+          tile.topRight,
+          { parkings ->
+            tilesToParking.value[tile] = parkings
+            _rectParkings.value +=
+                tilesToParking.value.filter { it.key in tilesToDisplay }.values.flatten()
```

Comment:
This increment value could just be `parkings` right?

Reply:
ooooooh that is so much faster and niceer, I feel dumb, I am gonna try